transition_lib/transition_api_lib.py

```diff
@@ -105,45 +49,50 @@ def build_headers():
         return headers
     
     @staticmethod
-    def get_token(username, password):
+    def get_token(username, password,url=None):
+        url = Transition.BASE_URL if url is None else url
         body = Transition.build_body(username, password)
-        response = requests.post(f"{Transition.BASE_URL}/token", json=body)
-        if response.status_code == 200:
-            Transition.config['credentials']['token'] = response.text
-
-        with open(Transition.config_path, 'w') as config_file:
-            Transition.config.write(config_file)
-
+        response = requests.post(f"{url}/token", json=body)
+        response.raise_for_status()
         return response.text
 
     @staticmethod
-    def get_paths():
-        headers = Transition.build_headers()
-        result = requests.get(f"{Transition.API_URL}/paths", headers=headers)
-        result.raise_for_status()
-        return result.json()
+    def get_paths(url=None, token=None):
+        url = Transition.API_URL if url is None else url
+        token = Transition.TOKEN if token is None else token            
```

Comment:
messemble c'est un peu redondant de mettre les 2 paramètres dans chaque méthode et ensuite de faire ces 2 lignes là partout... faudrait trouver un moyen de ne pas dupliquer ça partout j'ai l'impression, quitte à faire une méthode privée qui fait ces 2 lignes là (??)

Reply:
et c'est quoi le use case pour lequel un user voudrait changer de url ou de token à chaque requête? J'ai l'impression que demander au user de set ces 2 valeurs au début serait moins laid.

Peut-être qu'en fait on devrait switch vers une classe non statique et lorsqu'on initialise la classe on doit fournir le url et les credentials?

Reply:
y a pas de case. c'est juste pour donner l'option d'envoyer son token et url a chaque requête ou de le set once and for all au début pour une instance. Je sais pas si tu avais vu la discussion mais on a décidé de faire ça comme ça parce que en général les librairies en python te demandent pas de set des attributs mais les envoyer en paramètre à chaque fois, mais on a laissé les setters, faut juste documenter ça bien
also on les garde en paramètre parce que si c'est pour faire une seule requête on peut juste faire Transition.get_paths(url, token) pis ça se fait directement en une ligne sans avoir à créer / setter des objets..
est-ce que ça fait du sens ?

Reply:
Oui je vois!